### PR TITLE
validate ParseRangeString return value in EvaluateThreshold

### LINKED ISSUE #233: panic in EvaluateThreshold because of unvalidated ParseRangeString return value
ParseRangeString may return NIL on errors, but in EvaluateThreshold this is not checked will lead to a panic

`runtime error: invalid memory address or nil pointer dereference 
goroutine 36 [running]:
runtime/debug.Stack()
	C:/Users/user/go-sdk/go1.21.3/src/runtime/debug/stack.go:24 +0x6b
github.com/atc0005/go-nagios.(*Plugin).ReturnCheckResults(0xc0000c40e0)
	C:/local/progs/go-proj/src/check_hm/vendor/github.com/atc0005/go-nagios/nagios.go:283 +0x305
panic({0x1af6600?, 0x216afd0?})
	C:/Users/user/go-sdk/go1.21.3/src/runtime/panic.go:920 +0x290
github.com/atc0005/go-nagios.(*Plugin).EvaluateThreshold(0xc0000c40e0, {0xc0001f16c0, 0x1, 0x1})
	C:/local/progs/go-proj/src/check_hm/vendor/github.com/atc0005/go-nagios/range.go:153 +0xed
`
First 
https://github.com/atc0005/go-nagios/blob/7a6e9212a75d7cf063a1d0c787c74a4cb847f1a1/range.go#L151-L153
may panic here if CritcalThreshold object is nil

second:
https://github.com/atc0005/go-nagios/blob/7a6e9212a75d7cf063a1d0c787c74a4cb847f1a1/range.go#L160-L164

 may panic here if warningThreshold object is nil

### PR DESCRIPTION
**Pull request by Tommi2Day**
this PR adds checking of nil returns from ParseRangeString in EvaluateThreshold. it returns UNKNOWN state result and an error message.

second I added 2 tests to check this expected behavior

should fix #233

### CODE REVIEW

**Comment by atc0005** [comment]
@Tommi2Day thank you for submitting this PR. I should be able to take a look at this in the next few days and get back to you.

The addition of tests is much appreciated!

**Comment by Tommi2Day** [comment]
added fix for #235

File: range.go [hunk 1]
```diff
@@ -149,6 +150,11 @@ func (p *Plugin) EvaluateThreshold(perfData ...PerformanceData) error {
 		if perfData[i].Crit != "" {
 
 			CriticalThresholdObject := ParseRangeString(perfData[i].Crit)
+			if CriticalThresholdObject == nil {
+				err := fmt.Errorf("invalid critical threshold: %s ", perfData[i].Crit)
```
Comment (atc0005) [inline range.go:154]: Minor: We could adjust this in a follow-up PR, but wanted to note it while it was on my mind.

One of the approaches used thus far in this library is providing wrapped static errors. I don't know that is the *right* approach to use, but for consistency what do you think about adding to the sentinel error collection here:

https://github.com/atc0005/go-nagios/blob/7a6e9212a75d7cf063a1d0c787c74a4cb847f1a1/nagios.go#L72-L74

Something like:

```go
	// ErrInvalidRangeThreshold indicates that a given range threshold is not
	// in a supported format.
	ErrInvalidRangeThreshold = errors.New("invalid range threshold")
```

and line 154 becomes something like one of:

```go
err := fmt.Errorf("failed to parse critical range string %s: %w ", perfData[i].Crit, ErrInvalidRangeThreshold )
```

```go
err := fmt.Errorf("error parsing critical range string %s: %w ", perfData[i].Crit, ErrInvalidRangeThreshold )
```

File: range.go [hunk 2]
```diff
@@ -158,6 +164,11 @@ func (p *Plugin) EvaluateThreshold(perfData ...PerformanceData) error {
 
 		if perfData[i].Warn != "" {
 			warningThresholdObject := ParseRangeString(perfData[i].Warn)
+			if warningThresholdObject == nil {
+				err := fmt.Errorf("invalid warning threshold: %s ", perfData[i].Warn)
```
Comment (atc0005) [inline range.go:168]: Same feedback here as line 154.

**Review by atc0005 — CHANGES_REQUESTED** [review]
Overall, this is great. Many thanks for taking the time to report the problems and craft a set of changes to resolve them. The inclusion of test cases is especially appreciated.

I've highlighted a few spots for further discussion. Please let me know if I can clarify anything.

**Pull request closed without merging**